src/clarity-icons/clarity-icons-element.ts

```diff
@@ -147,7 +173,13 @@ ClarityIconElement.prototype._setAriaLabelledBy = function() {
     }
 };
 
-ClarityIconElement.prototype._injectTemplate = function() {
+ClarityIconElement.prototype._injectTemplate = function(shapeTemplate?: string) {
+    // Accepting the argument, shapeTemplate, will help us to update the shape template
+    // right before the injection.
+    if (shapeTemplate && shapeTemplate !== this.currentShapeTemplate) {
+        this.currentShapeTemplate = shapeTemplate;
+    }
+
     this.innerHTML = this.currentShapeTemplate;
```

Comment:
Should this and the following `if` be inside the `if` above so it only runs if a new template is provided?

Reply:
You mean like this?
```ts
 if (shapeTemplate && shapeTemplate !== this.currentShapeTemplate) {
       this.currentShapeTemplate = shapeTemplate;
       this.innerHTML = this.currentShapeTemplate;
}
```

Reply:
Right, if I understand it we don't want to reapply the innerHTML unless it changes.